internal/pkg/ecslogging/service.go

```diff
@@ -0,0 +1,88 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package ecslogging
+
+import (
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/copilot-cli/internal/pkg/aws/cloudwatchlogs"
+	"github.com/aws/copilot-cli/internal/pkg/term/log"
+)
+
+const (
+	logGroupNamePattern = "/copilot/%s-%s-%s"
+)
+
+type logGetter interface {
+	LogEvents(logGroupName string,
+		streamLastEventTime map[string]int64,
+		opts ...cloudwatchlogs.GetLogEventsOpts) (*cloudwatchlogs.LogEventsOutput, error)
+}
+
+// ServiceClient retrieves the logs of an Amazon ECS service.
+type ServiceClient struct {
+	logGroupName string
+	eventsGetter logGetter
+	w            io.Writer
+}
+
+// WriteLogEventsOpts wraps the parameters to call WriteLogEvents.
+type WriteLogEventsOpts struct {
+	Follow    bool
+	Limit     int
+	StartTime int64
+	EndTime   int64
+	// OnEvents is a handler that's invoked when logs are retrieved from the service.
+	OnEvents func(w io.Writer, logs []HumanJSONStringer) error
+}
+
+// NewServiceClient returns a ServiceClient configured against the input.
+func NewServiceClient(sess *session.Session, appName, envName, svcName string) *ServiceClient {
```

Comment:
What do you think of this comment instead to be more explicit?
```suggestion
// NewServiceClient returns a ServiceClient for the svc service under env and app.
// The logging client is initialized from the given sess session.
func NewServiceClient(sess *session.Session, app, env, svc string) *ServiceClient {
```